### PR TITLE
Internal user authz

### PR DESCRIPTION
**Pull request by camdencheek**
This updates code monitors and saved searches to run with the owning user ID. Additionally, it removes the possibility for code monitors and saved searches to be owned by an organization rather than a user. 

This updates the database layer to not reference the deprecated org_id columns on both tables, but does not go as far as changing the graphql API to not accept org IDs. However, it will now error if an org ID rather than a user is given. 

We are removing support for org-owned saved searches and code monitors because there is no clear way to handle permissions for running a query "as" an org. Currently, we run code monitors as site admin, which can leak information about the existence of search results for repos that a user does not have access to. This PR changes it so we run as the user who created the code monitor, so a user can't get information about any repos they don't already have access to.

### CODE REVIEW

**Comment by camdencheek** [comment]
@tsenart Still doing a bit of manual testing (since I have no idea how trustworthy these tests are), but tests are passing. Should be ready to merge tomorrow

**Comment by camdencheek** [comment]
Tested this locally, and any configuration through the UI seems to be working. I haven't tested just how robustly we handle bad input through the GraphQL API, but I'm not immediately concerned about that. I'll follow up with an update to the graphql layer

**Review by tsenart — APPROVED** [review]
LGTM. Would be good to leave context in this PR why we are removing org support.

File: cmd/frontend/internal/cli/http.go [hunk 1]
```diff
@@ -139,14 +140,24 @@ func newInternalHTTPHandler(schema *graphql.Schema, db database.DB, newCodeIntel
...
-func withInternalActor(h http.Handler) http.Handler {
+// for the internal endpoint), because internal requests can bypass repository permissions checks.
+func withActor(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		rWithActor := r.WithContext(actor.WithActor(r.Context(), &actor.Actor{Internal: true}))
+		var a actor.Actor
+
+		userID, err := strconv.ParseInt(r.Header.Get("X-Sourcegraph-User-ID"), 10, 32)
+		if err != nil || userID == 0 {
+			a.Internal = true
```
Comment (eseliger) [inline cmd/frontend/internal/cli/http.go:155]: It seems a bit dangerous that an invalid integer just silently causes internal actor privileges? Should we 400 instead?

Comment (tsenart) [reply cmd/frontend/internal/cli/http.go:155]: That was the previous default, but perhaps we can distinguish between the absence of the header and an invalid value.

Comment (eseliger) [reply cmd/frontend/internal/cli/http.go:155]: Yeah I think that'd be a good change

**Pull request merged**